modules/core/perf/perf_math.cpp

```diff
@@ -36,6 +36,242 @@ PERF_TEST_P(VectorLength, phase64f, testing::Values(128, 1000, 128*1024, 512*102
     SANITY_CHECK(angle, 5e-5);
 }
 
+// generates random vectors, performs Gram-Schmidt orthogonalization on them
+Mat randomOrtho(int rows, int ftype, RNG& rng)
```

Comment:
Gram-Schmidt is known to produce huge rounding errors. Use QR decomposition instead and remove the 'R' part.

Reply:
1. We don't need exact orhogonalization, just some random orthogonal matrix
2. I've checked its orthogonality (A^T*A-I as well as A*A^T-I) and its in acceptable epsilon bounds
3. The only QR decomposition we have is in `cv::hal` namespace, I'm not sure whether it's OK to use it in test even if it's exposed to public API